core/templates/pages/landing-pages/stewards-landing-page/stewards-landing-page.component.ts

```diff
@@ -16,133 +16,148 @@
-          }, {
-            text: 'Browse Lessons',
-            href: '/community-library'
-          }];
-        } else if (tabName === ctrl.TAB_NAME_VOLUNTEERS) {
-          // TODO(sll): Add "Get in Touch" link that points to contact form.
-          return [{
-            text: 'Browse Volunteer Opportunities',
-            href: 'https://www.oppiafoundation.org/volunteer'
-          }];
-        } else {
-          throw new Error('Invalid tab name: ' + tabName);
-        }
-      };
+  ngOnInit(): void {
+    // Set the initial tab name based on the URL; default to
+    // TAB_NAME_PARENTS.
+    let initialPathname = this.urlService.getPathname();
+    for (let urlPattern in this.URL_PATTERNS_TO_TAB_NAMES) {
+      if (initialPathname.indexOf(urlPattern) === 0) {
+        this.activeTabName = this.URL_PATTERNS_TO_TAB_NAMES[urlPattern];
+        break;
+      }
+    }
+    this.buttonDefinitions = this.getButtonDefinitions(this.activeTabName);
+    this.windowIsNarrow = this.isWindowNarrow(
+      this.windowDimensionsService.getWidth());
+    this.resizeSubscription = this.windowDimensionsService.getResizeEvent().
+      subscribe(evt => {
+        this.windowIsNarrow = this.isWindowNarrow(
+          this.windowDimensionsService.getWidth());
+      });
+  }
+
+  setActiveTabName(newActiveTabName: string): void {
+    this.activeTabName = newActiveTabName;
+    this.buttonDefinitions = this.getButtonDefinitions(newActiveTabName);
+  }
+
+  isActiveTab(tabName: string): boolean {
+    return this.activeTabName === tabName;
+  }
 
-      ctrl.getStaticImageUrl = function(imagePath) {
-        return UrlInterpolationService.getStaticImageUrl(imagePath);
-      };
+  getActiveTabNameInSingularForm(): string {
+    if (this.activeTabName === this.TAB_NAME_PARENTS) {
+      return 'Parent';
+    } else if (this.activeTabName === this.TAB_NAME_TEACHERS) {
+      return 'Teacher';
+    } else if (this.activeTabName === this.TAB_NAME_NONPROFITS) {
+      return 'Nonprofit';
+    } else if (this.activeTabName === this.TAB_NAME_VOLUNTEERS) {
+      return 'Volunteer';
+    } else {
+      throw new Error('Invalid active tab name: ' + this.activeTabName);
+    }
+  }
+
+  getButtonDefinitions(tabName: string): ButtonDefinition[] {
+    if (tabName === this.TAB_NAME_PARENTS ||
+        tabName === this.TAB_NAME_TEACHERS) {
+      return [{
+        text: 'Browse Lessons',
+        href: '/community-library'
+      }, {
+        text: 'Subscribe to our Newsletter',
+        href: 'https://eepurl.com/g5v9Df'
+      }];
+    } else if (tabName === this.TAB_NAME_NONPROFITS) {
+      return [{
+        text: 'Get Involved',
+        href: (
+          'https://www.oppiafoundation.org/partnerships#get-in-touch')
+      }, {
+        text: 'Browse Lessons',
+        href: '/community-library'
+      }];
+    } else if (tabName === this.TAB_NAME_VOLUNTEERS) {
+      // TODO(sll): Add "Get in Touch" link that points to contact form.
```

Comment:
While we are here, @seanlip does this still apply? If yes can you add a TODO issue (I have trouble understanding what does this mean exactly)?

Reply:
Hi @vojtechjelinek, thanks for asking. I think this doesn't apply and can be deleted.

Reply:
Line Removed